src/components/screens/Makai/LeveragerCard.tsx

```diff
@@ -0,0 +1,274 @@
+import { t } from '@lingui/macro'
+import { BigNumber, valueToBigNumber } from '@starlay-finance/math-utils'
+import { StaticImageData } from 'next/image'
+import { FC, ReactNode, useEffect, useMemo, useState } from 'react'
+import { Image } from 'src/components/elements/Image'
+import { asStyled } from 'src/components/hoc/asStyled'
+import { TooltipMessage } from 'src/components/parts/ToolTip'
+import { useLeverager } from 'src/hooks/contracts/useLeverager'
+import { useLdotApy } from 'src/hooks/useLdotApy'
+import { darkGray, primary, purple } from 'src/styles/colors'
+import { fontWeightMedium, fontWeightSemiBold } from 'src/styles/font'
+import { AssetMarketData, AssetSymbol } from 'src/types/models'
+import { EthereumAddress } from 'src/types/web3'
+import { BN_ZERO, formatAmt } from 'src/utils/number'
+import styled from 'styled-components'
+
+type CardProps = {
+  collateralAsset?: AssetMarketData | undefined
+  icon: StaticImageData
+  borrowApy: BigNumber
+  symbol: AssetSymbol
+  balance: BigNumber
+  onClick: VoidFunction
+}
+
+export const LeveragerCard = asStyled<CardProps>(
+  ({
+    collateralAsset,
+    icon,
+    borrowApy,
+    balance,
+    symbol,
+    onClick,
+    className,
+  }) => {
+    const { getLTV } = useLeverager()
+    const [maxLeverage, setMaxLeverage] =
+      useState<number | undefined>(undefined)
+    const { apy } = useLdotApy()
+
+    const maxApy = useMemo(() => {
+      const maxApy =
+        (maxLeverage || 0) *
+        (apy -
+          borrowApy.toNumber() +
+          (collateralAsset?.depositAPY || BN_ZERO).toNumber()) *
+        100
+      return maxApy > 0 ? maxApy : 0
```

Comment:
I know that in general there can be cases where APY is negative, but isn't it a misstatement to set it to 0 if it is negative?
I don't know exactly how this feature works, so if it is in fact 0, it doesn't matter.